### PR TITLE
Optional vm type and region

### PR DESCRIPTION
**Pull request by aguillenv**
Related issue: https://github.com/alantech/meta/issues/183

Related PRs:
- https://github.com/alantech/alan/pull/506
- https://github.com/alantech/deploy/pull/98

### CODE REVIEW

**Review by cdmistman — APPROVED** [review]

File: anycloud/cli/src/deploy/mod.rs [hunk 1]
```diff
@@ -524,14 +528,22 @@ pub async fn add_deploy_config() {
...
-      .with_prompt("Region name")
-      .interact_text()
-      .unwrap();
-    let vm_type: String = Input::with_theme(&ColorfulTheme::default())
-      .with_prompt("Virtual machine type")
-      .interact_text()
-      .unwrap();
+    let mut region = None;
+    if confirm_prompt(
+      "Do you want to add a region name to this Deploy Config?",
```
Comment (dfellis) [inline anycloud/cli/src/deploy/mod.rs:533]: Maybe `"Do you want to choose a specific region for this Deploy Config?"`

Comment (dfellis) [reply anycloud/cli/src/deploy/mod.rs:533]:
```suggestion
      "Do you want to choose a specific region for this Deploy Config?",
```

File: anycloud/cli/src/deploy/mod.rs [hunk 2]
```diff
@@ -524,14 +528,22 @@ pub async fn add_deploy_config() {
...
+      false,
+    ) {
+      let input_region: String = input_prompt("Region name");
+      region = Some(input_region);
+    };
+    let mut vm_type = None;
+    if confirm_prompt(
+      "Do you want to add a virtual machine type to this Deploy Config?",
```
Comment (dfellis) [inline anycloud/cli/src/deploy/mod.rs:541]:
```suggestion
      "Do you want to select which virtual machine type to use for this Deploy Config?",
```

File: anycloud/cli/src/deploy/mod.rs [hunk 3]
```diff
@@ -633,16 +645,46 @@ pub async fn edit_deploy_config() {
...
-    let vm_type: String = Input::with_theme(&ColorfulTheme::default())
-      .with_prompt("Virtual machine type")
-      .with_initial_text(config.vmType.to_string())
-      .interact_text()
-      .unwrap();
+    let mut region = None;
+    let mut vm_type = None;
+    if let Some(reg) = &config.region {
+      if confirm_prompt(
+        "Do you want to edit the region name for this Deploy Config?",
```
Comment (dfellis) [inline anycloud/cli/src/deploy/mod.rs:652]:
```suggestion
        "Do you want to edit the region to use for this Deploy Config?",
```

File: anycloud/cli/src/deploy/mod.rs [hunk 4]
```diff
@@ -633,16 +645,46 @@ pub async fn edit_deploy_config() {
...
+        true,
+      ) {
+        let input_region: String = input_prompt("Region name");
+        region = Some(input_region);
+      } else {
+        region = Some(reg.to_string());
+      }
+    } else {
+      if confirm_prompt(
+        "Do you want to add a region name to this Deploy Config?",
```
Comment (dfellis) [inline anycloud/cli/src/deploy/mod.rs:662]:
```suggestion
        "Do you want to choose a specific region for this Deploy Config?",
```

File: anycloud/cli/src/deploy/mod.rs [hunk 5]
```diff
@@ -633,16 +645,46 @@ pub async fn edit_deploy_config() {
...
+        true,
+      ) {
+        let input_vm_type: String = input_prompt("Virtual machine type");
+        vm_type = Some(input_vm_type);
+      } else {
+        vm_type = Some(vm_t.to_string());
+      }
+    } else {
+      if confirm_prompt(
+        "Do you want to add a virtual machine type to this Deploy Config?",
```
Comment (dfellis) [inline anycloud/cli/src/deploy/mod.rs:681]:
```suggestion
        "Do you want to select which virtual machine type to use for this Deploy Config?",
```

**Review by dfellis — APPROVED** [review]
Many suggestions on prompt text.

**Review by dfellis — APPROVED** [review]

**Review by depombo — APPROVED** [review]

**Pull request merged**